src/compas_rhino/geometry/surface.py

```diff
@@ -4,354 +4,190 @@
 
 import Rhino
 import compas_rhino
-
+from compas.utilities import memoize
 from compas.datastructures import Mesh
-from compas.geometry import angle_vectors
-from compas.geometry import distance_point_point
-from compas.utilities import geometric_key
-
-from compas_rhino.geometry._geometry import BaseRhinoGeometry
-
-
-__all__ = ['RhinoSurface']
+from compas.datastructures import meshes_join
+# from compas.geometry import angle_vectors
+# from compas.geometry import distance_point_point
+# from compas.utilities import geometric_key
+from ..conversions import point_to_compas
+from ._geometry import BaseRhinoGeometry
 
 
 class RhinoSurface(BaseRhinoGeometry):
     """Wrapper for Rhino surface objects."""
 
-    def __init__(self):
-        super(RhinoSurface, self).__init__()
-
     @classmethod
-    def from_geometry(cls):
-        raise NotImplementedError
+    def from_geometry(cls, geometry):
+        """Construct a surface wrapper from an existing geometry object.
+
+        Parameters
+        ----------
+        geometry : :class:`Rhino.Geometry.Surface`
+            The geometry object defining a surface.
+
+        Returns
+        -------
+        :class:`RhinoSurface`
+            The Rhino surface wrapper.
+        """
+        if not isinstance(geometry, Rhino.Geometry.Surface):
+            raise NotImplementedError
```

Comment:
`NotImplementedError` or just a `TypeError`? because I don't think we'll ever implement a `RhinoSurface` conversion from anything that is not a surface..

Reply:
+1 for `TypeError`